### ISSUE TITLE
CKAN 2.9.4 cannot load JS translations when using Python 2.7

### ISSUE THREAD

**Issue by frafra** [ckan/ckan#6531]
This code is not Python 2.7 compatible:

https://github.com/ckan/ckan/blob/6731c5a821a6a5f4bdaa20f4e793e0b6ba44f823/ckan/views/api.py#L482

In Python 2, `encoding` is not a valid argument for the `open` function, but it is for the `json.load`.

Here is the commit which introduced the regression: https://github.com/ckan/ckan/commit/a280cf123748eb548a64242033aa41c0c67ee256

```python-traceback
2021-11-11 15:42:28,678 ERROR [ckan.config.middleware.flask_app] 'encoding' is an invalid keyword argument for this function
Traceback (most recent call last):
  File "/usr/lib/ckan/venv/local/lib/python2.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/ckan/venv/local/lib/python2.7/site-packages/flask/app.py", line 1935, in dispatch_request
   return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/ckan/venv/src/ckan/ckan/views/api.py", line 482, in i18n_js_translations
    translations = json.load(open(source, u'r', encoding=u'utf-8'))
TypeError: 'encoding' is an invalid keyword argument for this function
```

It seems to have been fixed partially already in another file: https://github.com/ckan/ckan/commit/b2b92a478f9a96c908e214d33d7d8141c73343c7

Related issue: #6051

**Comment by Zharktas** [comment]
The issue with relative import in lib directory still needs to be fixed.

**Comment by frafra** [comment]
The `io.py` has been renamed to `io_.py` on master.

Should that change be back ported too? https://github.com/ckan/ckan/commit/96693f22b14581de23ffcf28022ad90b974367d6

**Comment by frafra** [comment]
I am unable to reproduce on `dev-v2.9`. Fully fixed then?

**Comment by amercader** [comment]
AFAICT this has been fixed in 2.9

**Comment by stefina** [comment]
We have applied this patch for one of our instances running 2.9.4 and no longer get this error, so this is indeed the fix :)
